examples/error_estimates_forces.jl

```diff
@@ -0,0 +1,182 @@
+# # Practical error bounds for the forces
+#
+# This is a simple example showing how to compute error estimates for the forces on a
+# ``{\rm TiO}_2`` molecule, from which we can either compute asymptotically valid error
+# bounds or increase the precision on the computation of the forces.
+#
+# The strategy we follow is described with more details in[^CDKL2021] and we
+# will use in comments the density matrices framework. We will also needs
+# operators and functions from
+# [`src/scf/newton.jl`](https://dftk.org/blob/master/src/scf/newton.jl).
+#
+# [^CDKL2021]:
+#     E. Cancès, G. Dusson, G. Kemlin, and A. Levitt
+#     *Practical error bounds for properties in plane-wave electronic structure
+#     calculations* Preprint, 2021. [arXiv](https://arxiv.org/abs/2111.01470)
+using DFTK
+using LinearAlgebra
+using ForwardDiff
+using LinearMaps
+using IterativeSolvers
+
+# ## Setup
+# We setup manually the ``{\rm TiO}_2`` configuration from
+# [Materials Project](https://materialsproject.org/materials/mp-2657/).
+Ti = ElementPsp(:Ti, psp=load_psp("hgh/lda/ti-q4.hgh"))
+O = ElementPsp(:O, psp=load_psp("hgh/lda/o-q6.hgh"))
+atoms = [Ti => [[0.5, 0.5, 0.5],
+                [0.0, 0.0, 0.0]],
+         O  => [[0.19542, 0.80458, 0.5],
+                [0.80458, 0.19542, 0.5],
+                [0.30458, 0.30458, 0.0],
+                [0.69542, 0.69542, 0.0]]]
+lattice = [[8.79341  0.0      0.0];
+           [0.0      8.79341  0.0];
+           [0.0      0.0      5.61098]];
+
+# We apply a small displacement to one of the ``\rm Ti`` atoms to get nonzero forces.
+el, pos = atoms[1]
+atoms[1] = Pair(el, pos .+ [[-0.22, 0.28, 0.35] / 10, [0, 0, 0]]);
+# We build a model with one k-point only, not too high `Ecut_ref` and small tolerance to
+# limit computational time. These parameters can be increased for more precise
+# results.
+model = model_LDA(lattice, atoms)
+kgrid = [1, 1, 1]
+Ecut_ref = 35
+basis_ref = PlaneWaveBasis(model; Ecut=Ecut_ref, kgrid)
+tol = 1e-8;
+
+# ## Computations
+# We compute the reference solution ``P_*`` from which we will compute the
+# references forces.
+scfres_ref = self_consistent_field(basis_ref, tol=tol, callback=info->nothing)
+ψ_ref, occupation = DFTK.select_occupied_orbitals(basis_ref, scfres_ref.ψ, scfres_ref.occupation)
+# We compute a variational approximation of the reference solution with
+# smaller `Ecut`. Be careful to choose `Ecut` not too close to `Ecut_ref`.
+# /!\ The current choice `Ecut_ref=35` is such that the reference solution is
+# not converged and `Ecut=15` is such that the asymptotic regime
+# (crucial to validate the approach) is barely established.
+Ecut = 15
+basis = PlaneWaveBasis(model; Ecut=Ecut, kgrid)
+scfres = self_consistent_field(basis, tol=tol, callback=info->nothing)
+ψr = DFTK.transfer_blochwave(scfres.ψ, basis, basis_ref)
```

Comment:
Maybe add that the ψr, ρr and Er quantities are basically the quantities computed by the crude SCF extended to the reference basis.